sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -285,6 +291,11 @@ class FileSystemBasedCheckpointFileManager(path: Path, hadoopConf: Configuration
     case _: LocalFileSystem | _: RawLocalFileSystem => true
     case _ => false
   }
+
+  override def createCheckpointDirectory(): Path = {
+    fs.mkdirs(path, FsPermission.getDirDefault)
+    fs.makeQualified(path)
```

Comment:
shall we qualify first before calling `mkdirs`? Or it doesn't matter?

Reply:
Before this change, it is first qualified and then calls `mkdirs`. Let's follow up original behavior.